test/document.test.js

```diff
@@ -11798,6 +11798,37 @@ describe('document', function() {
     await doc.save();
     assert.equal(doc.data.sub, 'sub');
   });
+
+  it('handles maps when applying defaults to nested paths (gh-12220)', async function() {
+    const nestedSchema = new mongoose.Schema({
+      1: {
+        type: Number,
+        default: 0
+      }
+    });
+
+    const topSchema = new mongoose.Schema({
+      nestedPath1: {
+        mapOfSchema: {
+          type: Map,
+          of: nestedSchema
+        }
+      }
+    });
+
+    const Test = db.model('Test', topSchema);
+
+    const data = {
+      nestedPath1: {
+        mapOfSchema: {
+          // 2022: { 1: 0 }, // this data does not affect the error
```

Comment:
Can we rephrase this comment? I'm not sure what it means or if we need it.

Reply:
it was part of the original issue that i had reduced down (reproduction code), where that extra property did not affect the error, but i did not remove it to show the level of nesting required